src/loki/worker.cc

```diff
@@ -262,6 +267,43 @@ void loki_worker_t::set_interrupt(const std::function<void()>* interrupt_functio
   reader->SetInterrupt(interrupt);
 }
 
+// Check if total arc distance exceeds the max distance limit for disable_hierarchy_pruning.
+// If true, add a warning and set the disable_hierarchy_pruning costing option to false.
+void loki_worker_t::check_hierarchy_distance(Api& request, bool pair_wise) {
+  auto& options = *request.mutable_options();
+  auto costing_options = options.mutable_costings()->find(options.costing_type());
+  if (!costing_options->second.options().disable_hierarchy_pruning()) {
+    return;
+  }
+
+  bool max_distance_exceeded = false;
+  if (pair_wise) {
```

Comment:
instead of passing the bool we can simply check `request.Options.action()` and see if its the `sources_to_targets` type

Reply:
done